SkillTrack-Team/skill-track
# Feat/set up db

# Description
Refactored server boilerplate and set up database connection

## Type of change


Fixes # (issue) Set up database

## Changes Made
- Restructured the server folder boilerplate for clearer organization
- Added types folder 
- Installed nodemon for automatic server restarts during development
- Connected the server to a PostgreSQL Elephant SQL database
- Created tables (users, applications, interview_rounds, comments, interview questions) in the Elephant SQL console
- Added database connection testing in UserController and UserRouter

# Testing Instructions

1. Navigate to the **server** directory
2. Make sure there is a **.env** file that includes **PG_URI**
3. In the terminal, run **npm run dev**
4. In postman, make a get request to **localhost:3000/api/users** or in the browser navigate to the same url
5. Make sure you are receiving the response `[
    {
        "id": 1,
        "first_name": "testFirstName",
        "last_name": "testLastName",
        "linkedin_id": "testLinkedinId"
    }
]`

# Checklist:

- [x] I have followed the coding style and guidelines of this project.
- [x] I have reviewed my own code to ensure its quality.
- [ ] I have added relevant comments and documentation where necessary.
- [ ] I have updated/added necessary tests for the changes made.
- [ ] I have rebased my branch on top of the latest main (or other target) branch.
